recipe/build.sh

```diff
@@ -34,38 +11,34 @@ fi
 export PYTHON="$PYTHON"
 export PYTHON_LDFLAGS="$PREFIX/lib"
 export LDFLAGS="$LDFLAGS -L$PREFIX/lib -Wl,-rpath,$PREFIX/lib"
-export CFLAGS="$CFLAGS -fPIC -I$PREFIX/include"
+# The `ACCEPT_USE_OF_DEPRECATED_PROJ_API_H` is a temporary solution and won't work with proj4 7.
+export CFLAGS="$CFLAGS -fPIC -I$PREFIX/include -DACCEPT_USE_OF_DEPRECATED_PROJ_API_H=1"
+export CPPFLAGS="-DACCEPT_USE_OF_DEPRECATED_PROJ_API_H=1 ${CPPFLAGS}"
```

Comment:
@conda-forge/magics team looks like this was not enough to fix the build here. Not sure if magics requires more patching.

Reply:
@ocefpaf I believe ECMWF are aware that Magics needs some work to make it compatible with Proj4 v6 and that it's in-progress. I would leave this PR for now until it's sorted out!

Reply:
>  I would leave this PR for now until it's sorted out!

For migration purposes closing this PR is fine and the bot knows to move on. For `magics`, once it is OK to move to the new proj, all you have to do re-render.